waiter/src/waiter/token.clj

```diff
@@ -434,7 +434,8 @@
       (when (not-empty unknown-keys)
         (throw (ex-info (str "Unsupported key(s) in token: " (str (vec unknown-keys)))
                         {:status http-400-bad-request :token token :log-level :warn}))))
-    (let [service-parameter-with-token-defaults (attach-token-defaults-fn new-service-parameter-template)]
+    (let [service-parameter-with-token-defaults (attach-service-defaults-fn new-service-parameter-template)
```

Comment:
Did we audit all other usages of `attach-token-defaults` to make sure this mistake isn't repeated elsewere?

Reply:
Also, this should be renamed <i>service-parameter-with-<b>service</b>-defaults</i>.

Reply:
The [previous PR which wanted to fix the bug](https://github.com/twosigma/waiter/pull/1197/files) introduced `attach-token-defaults` into the file. It has now been replaced with uses of `attach-service-defaults` in this file.

Also, renamed variable to `service-parameter-with-service-defaults`.

Reply:
> Did we audit all other usages of attach-token-defaults to make sure this mistake isn't repeated elsewhere?

I audited all uses in this file where we introduced the function as part of a [recent PR](https://github.com/twosigma/waiter/pull/1197/files).

I also checked the other uses. In all but one usage, the function is passed around as a pair to `attach-service-defaults`. The only single pass of `attach-token-defaults` is to compute token GC time which relies on token only parameters: `fallback-period-secs` and `stale-timeout-mins` in `waiter.service-description/source-tokens->gc-time-secs`.